cloud/scope/powervs_cluster.go

```diff
@@ -157,3 +292,1545 @@ func (s *PowerVSClusterScope) PatchObject() error {
+		Target:         &target,
+		ResourceGroup:  &resourceGroupID,
+		ResourcePlanID: pointer.String(cosResourcePlanID),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return serviceInstance, nil
+}
+
+// getResourceGroupID retrieving id of resource group.
+func (s *PowerVSClusterScope) getResourceGroupID() (string, error) {
+	rmv2, err := resourcemanagerv2.NewResourceManagerV2(&resourcemanagerv2.ResourceManagerV2Options{
+		Authenticator: s.session.Options.Authenticator,
+	})
+	if err != nil {
+		return "", err
+	}
+	if rmv2 == nil {
+		return "", fmt.Errorf("unable to get resource controller")
+	}
+	resourceGroup := s.ResourceGroup()
+	rmv2ListResourceGroupOpt := resourcemanagerv2.ListResourceGroupsOptions{Name: resourceGroup, AccountID: &s.session.Options.UserAccount}
+	resourceGroupListResult, _, err := rmv2.ListResourceGroups(&rmv2ListResourceGroupOpt)
+	if err != nil {
+		return "", err
+	}
+
+	if resourceGroupListResult != nil && len(resourceGroupListResult.Resources) > 0 {
+		rg := resourceGroupListResult.Resources[0]
+		resourceGroupID := *rg.ID
+		return resourceGroupID, nil
+	}
+
+	err = fmt.Errorf("could not retrieve resource group id for %s", *resourceGroup)
+	return "", err
+}
+
+// getVPCRegion returns region associated with VPC zone.
+func (s *PowerVSClusterScope) getVPCRegion() *string {
+	if s.IBMPowerVSCluster.Spec.VPC != nil {
+		return s.IBMPowerVSCluster.Spec.VPC.Region
+	}
+	// if vpc region is not set try to fetch corresponding region from power vs zone
+	if s.Zone() == nil {
+		return nil
+	}
+	region := endpoints.ConstructRegionFromZone(*s.Zone())
+	vpcRegion, err := genUtil.VPCRegionForPowerVSRegion(region)
+	if err != nil {
+		s.Error(err, fmt.Sprintf("failed to fetch vpc region associated with powervs region %s", region))
+		return nil
+	}
+	return &vpcRegion
+}
+
+// fetchVPCCRN returns VPC CRN
+func (s *PowerVSClusterScope) fetchVPCCRN() (*string, error) {
+	vpcDetails, _, err := s.IBMVPCClient.GetVPC(&vpcv1.GetVPCOptions{
+		ID: s.GetVPCID(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return vpcDetails.CRN, nil
+}
+
+// fetchPowerVSServiceInstanceCRN returns Power VS service instance CRN.
+func (s *PowerVSClusterScope) fetchPowerVSServiceInstanceCRN() (*string, error) {
+	serviceInstanceID := s.GetServiceInstanceID()
+	pvsDetails, _, err := s.ResourceClient.GetResourceInstance(&resourcecontrollerv2.GetResourceInstanceOptions{
+		ID: &serviceInstanceID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return pvsDetails.CRN, nil
+}
+
+// TODO(karthik-k-n): Decide on proper naming format for services
+
+// GetServiceName returns name of given service type from spec or generate a name for it.
+func (s *PowerVSClusterScope) GetServiceName(resourceType ResourceType) *string {
+	switch resourceType {
+	case ServiceInstance:
+		if s.ServiceInstance() == nil || s.ServiceInstance().Name == nil {
+			return pointer.String(fmt.Sprintf("%s-serviceInstance", s.InfraCluster().GetName()))
```

Comment:
Regarding the suffix, sometime back had a discussion with IPI team and decided to use common suffix, for powervs and other few resources which hypershift infra creation also follows currently. Also keeping longer suffix would make difficult to find the cluster resources in ibm cloud portal. Keeping it short would be better IMO.
powervs - pvs
vpc - vpc
vpc subnet - vpc-sn
transit gateway - tg
cos - image-registry - this alone shared by ibmcloud platform as well
Please see whether using this would make sense!

Reply:
We will handle this in next phase.